subsys/mgmt/osdp/src/osdp_cp.c

```diff
@@ -114,10 +114,13 @@ int osdp_extract_address(int *address)
 	return (pd_offset == CONFIG_OSDP_NUM_CONNECTED_PD) ? 0 : -1;
 }
 
-static inline void assert_buf_len(int need, int have)
+static inline bool check_buf_len(int need, int have)
 {
-	__ASSERT(need < have, "OOM at build command: need:%d have:%d",
-		 need, have);
+	if (need >= have) {
```

Comment:
Why `need > have` is not sufficient (negative) check condition?
If there is some assumption for usage of the last byte buffer then a small comment will be very appreciated here (the same for PD part).

Reply:
`need > have` would be fine here. I was retaining the existing condition to have no functional change in this commit. The condition in the `__ASSERT()` comes from a time when the command/reply ID (1 byte) was not part of `need`.